src/libsyntax/parse/lexer/mod.rs

```diff
@@ -1439,8 +1439,15 @@ impl<'a> StringReader<'a> {
             'r' => {
                 let start_bpos = self.pos;
                 self.bump();
-                let mut hash_count: u16 = 0;
+                let mut hash_count: u8 = 0;
                 while self.ch_is('#') {
+                    if hash_count == 255 {
+                        let last_bpos = self.pos;
+                        self.fatal_span_(start_bpos,
+                                         last_bpos,
+                                         "too many `#` characters surrounding a raw string; \
```

Comment:
The wording could be a little more concise here. Maybe:
```
too many `#` symbols: raw strings may be delimited by up to 255 `#` symbols
```
(and same for byte strings).